tools/bin/commands/util/operator_funcs

```diff
@@ -20,25 +37,52 @@ dep_ensure() {
         dep ensure -vendor-only -v
     else
         echo "Running 'dep ensure' with $GO_BUILDER_IMAGE"
-        [ -d "dep-cache" ] || mkdir dep-cache
+        required_dirs
         local extra_opts=""
         if [ -n "$as_user" ]; then
             extra_opts="$extra_opts -u $as_user"
         fi
         docker run -w /gopath/src/github.com/syndesisio/syndesis/install/operator \
+                   -u $(id -u):$(id -g) \
                    $extra_opts \
+                   -v "$DOCKER_GOPATH_VOLUME":/gopath:z \
+                   -v "$DOCKER_CACHE_VOLUME":/.cache:z \
                    -v $operator_dir/dep-cache:/gopath/pkg/dep:z \
                    -v $operator_dir:/gopath/src/github.com/syndesisio/syndesis/install/operator:z \
                    $GO_BUILDER_IMAGE \
                    dep ensure -vendor-only -v
     fi
 }
 
+required_dirs() {
+    require_dir "$operator_dir/dep-cache"
+    require_dir "$DOCKER_CACHE_VOLUME"
+    require_dir "$DOCKER_GOPATH_VOLUME/pkg/dep"
+    require_dir "$DOCKER_GOPATH_VOLUME/src/github.com/syndesisio/syndesis/install/operator"
+
+    CLEANUP_DIRS+=($DOCKER_GOPATH_VOLUME)
```

Comment:
You shouldn't remove cache dirs as they considerably will increase the build time. Maybe an option `--clean` would be better which would remove that dirs _before_ calling.

Reply:
When you add here $DOCKER_GOPATH_VOLUME here to the CLEANUP_DIRS then it won't get delete when you use --local (but it's still created)